AndroidNativeLoginTemplate/app/src/main/java/com/salesforce/androidnativelogintemplate/NativeLogin.kt

```diff
@@ -0,0 +1,327 @@
+                    // If we don't SalesforceSDKManager.getInstance() will complain it has not be setup.
+                    LoginView(
+                        // Pass this inline login function to be executed within the composable.
+                        login = { username, password -> Boolean
+                            // Call login and handle results.
+                            when (val result = nativeLoginManager.login(username, password)) {
+                                NativeLoginResult.InvalidUsername -> {
+                                    Toast.makeText(baseContext, result.name,Toast.LENGTH_LONG).show()
+                                    return@LoginView false
+                                }
+                                NativeLoginResult.InvalidPassword -> {
+                                    Toast.makeText(baseContext, result.name,Toast.LENGTH_LONG).show()
+                                    return@LoginView false
+                                }
+                                NativeLoginResult.InvalidCredentials -> {
+                                    Toast.makeText(baseContext, result.name,Toast.LENGTH_LONG).show()
+                                    return@LoginView false
+                                }
+                                NativeLoginResult.UnknownError -> {
+                                    Toast.makeText(baseContext, result.name, Toast.LENGTH_LONG).show()
+                                    return@LoginView false
+                                }
+                                NativeLoginResult.Success -> {
+                                    finish()
+                                }
+                            }
+
+                            return@LoginView true
+                        },
+                        handleWebviewFallbackResult,
+                        nativeLoginManager.getFallbackWebAuthenticationIntent(),
+                        nativeLoginManager.shouldShowBackButton,
+                        backAction = { finish() },
+                    )
+                }
+            }
+        )
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            onBackInvokedDispatcher.registerOnBackInvokedCallback(
+                OnBackInvokedDispatcher.PRIORITY_DEFAULT
+            ) {
+                if (nativeLoginManager.shouldShowBackButton) {
+                    when (SalesforceSDKManager.getInstance().userAccountManager.authenticatedUsers) {
+                        null -> moveTaskToBack(false)
+                        else -> finish()
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun LoginView(
+    login: suspend (String, String) -> Boolean,
+    handleWebviewFallbackResult: ActivityResultLauncher<Intent>? = null,
+    webviewLoginIntent: Intent? = null,
+    shouldShowBack: Boolean = false,
+    backAction: () -> Unit,
+) {
```

Comment:
This could be a lot cleaner if we don't care about the Jetpack Compose Preview, but I think it is valuable to provide an example that utilizes it.